File: domain/trainings-domain/src/main/java/com/fit/fitndflow/app/domain/trainings/repository/TrainingRepository.java
package com.fit.fitndflow.app.domain.trainings.repository;

import com.fit.fitndflow.app.domain.common.models.CategoryModel;
import com.fit.fitndflow.app.domain.common.models.ExerciseModel;
import com.fit.fitndflow.app.domain.common.models.SerieModel;
import com.fit.fitndflow.app.domain.common.repository.CommonRepository;
import java.util.List;

public interface TrainingRepository extends CommonRepository {
    ExerciseModel addNewSerie(int reps, double weight, int exerciseId) throws Exception;
    ExerciseModel modifySerie(int serieId, int reps, double weight) throws Exception;
    ExerciseModel deleteSerie(int serieId) throws Exception;
    List<CategoryModel> getTrainingListAndUpdateCache(String date) throws Exception;
    List<CategoryModel> updateCurrentTrainingListCache() throws Exception;
    List<SerieModel> getSerieListOfExerciseAdded(int exerciseid) throws Exception;
}

File: domain/trainings-domain/src/main/java/com/fit/fitndflow/app/domain/trainings/repository/TrainingRepository.kt
package com.fit.fitndflow.app.domain.trainings.repository

import com.fit.fitndflow.app.domain.common.models.CategoryModel
import com.fit.fitndflow.app.domain.common.models.ExerciseModel
import com.fit.fitndflow.app.domain.common.models.SerieModel
import com.fit.fitndflow.app.domain.common.repository.CommonRepository

interface TrainingRepository : CommonRepository {
    suspend fun addNewSerie(reps: Int, weight: Double, exerciseId: Int): ExerciseModel?
    suspend fun modifySerie(serieId: Int, reps: Int, weight: Double): ExerciseModel?
    suspend fun deleteSerie(serieId: Int): ExerciseModel?
    suspend fun getTrainingListAndUpdateCache(date: String): List<CategoryModel>
    suspend fun updateCurrentTrainingListCache(): List<CategoryModel>
    suspend fun getSerieListOfExerciseAdded(exerciseid: Int): List<SerieModel>
}
